### ISSUE TITLE
Problem with sample_shape

### ISSUE THREAD

**Issue by guilhermebs** [pyro-ppl/numpyro#333]
There seems to be a problem with sample_shape in version 0.2. Apparently the object that is returned by the sample primitive cannot be indexed. Also calls to `len()` and `.shape` do not return the expected results. It used to work in version 0.1

```python
import numpy as onp
import jax.numpy as np
from jax import random, vmap
import numpyro
import numpyro.distributions as dist
from numpyro.mcmc import MCMC, NUTS

# Create some data
x = np.linspace(0, 1)
slope = 3.
intercept = 1.
sigma = 0.2
y = slope*x + intercept + onp.random.normal(0., sigma, len(x))

# Model without sample_shape
def model1(x, y=None):
    slope = numpyro.sample('slope', dist.Normal(0., 10.))
    intercept = numpyro.sample('intercept', dist.Normal(0., 10.))
    sigma = numpyro.sample('sigma', dist.Exponential(1.))
    numpyro.sample('y', dist.Normal(slope*x + intercept, sigma), obs=y)

# Model with sample_sape
def model2(x, y=None):
    coefficients = numpyro.sample('slope', dist.Normal(0., 10.), sample_shape=(2, 1))
    sigma = numpyro.sample('sigma', dist.Exponential(1.))
    numpyro.sample('y', dist.Normal(coefficients[0]*x + coefficients[1], sigma), obs=y)


# Run NUTS
rng = random.PRNGKey(0)
rng, rng_ = random.split(rng)

num_warmup, num_samples = 1000, 2000

# Model 1
kernel = NUTS(model1)
mcmc = MCMC(kernel, num_warmup, num_samples)
mcmc.run(rng_, x=x, y=y) # Runs fine
mcmc.print_summary()
samples_1 = mcmc.get_samples()

# Model 2
kernel = NUTS(model2)
mcmc = MCMC(kernel, num_warmup, num_samples)
mcmc.run(rng_, x=x, y=y)
# IndexError: Too many indices for array: 1 non-None/Ellipsis indices for dim 0.
mcmc.print_summary()
samples_1 = mcmc.get_samples()
```

**Comment by fehiepsi** [comment]
@guilhermebs Thanks so much for a clear bug report! This is a regression when we tried to find the initial points. We fixed this issue at #332. As a temporary workaround, you can replace
```
coefficients = numpyro.sample('slope', dist.Normal(0., 10.), sample_shape=(2, 1))
```
by
```
coefficients = numpyro.sample('slope', dist.Normal(np.zeros((2, 1)), 10.))
```
:)

**Comment by guilhermebs** [comment]
Happy to help! I'm really enjoying the numpyro.
I've done some additional testing. i was not using 0.1.0, but rather  commit 91e2fae5b03b21007d858d35292c0dfefb4b111d

In my actual code I'm using Uniform distribution instead of normal distribution. Is there a workaround for this case as well?

**Comment by fehiepsi** [comment]
Yes, I just tested that the workaround works for Uniform distribution as well
```
coefficients = numpyro.sample('slope', dist.Uniform(np.zeros((2, 1)), 10.))
```

**Comment by neerajprad** [comment]
Thanks for giving NumPyro a try. As mentioned by @fehiepsi, this should be fixed by #332 which is merged into the master branch. Please feel free to reopen this if you face any issues!
